### PR TITLE
Add tests for permissions for variant overrides

### PR DESCRIPTION
**Pull request by kristinalim**
#### What? Why?

Relates to #2739 (Verify before closing issue)

This PR adds tests outlining the rules of permissions for variant overrides.

```
permissions for variant overrides
  when admin
    should have permission
  when user of the producer
    should not have permission
  when user of the distributor
    should not have permission
  when user of the distributor which is also the producer
    should have permission
  when owner of the distributor with add_to_order_cycle permission to the producer
    should not have permission
  when owner of the enterprise with create_variant_overrides permission to the producer
    should not have permission
    when the enterprise is not a distributor
      should not have permission
```

#### What should we test?
Build is green and all these new specs are green.

#### Release notes

Changelog Category: Added
Improved tests around user access to manage inventories.

#### How is this related to the Spree upgrade?
Not related.

### CODE REVIEW

**Comment by mkllnk** [comment]
@kristinalim Moving this to "In Dev" as it doesn't seem ready for code review.

**Comment by HugsDaniel** [comment]
All tests are green on both master and 0312bf93dde2bb9e976ccfaae954c991d6a0852f (which is the current version of Canada instance).

**Comment by mkllnk** [comment]
@kristinalim What's the state of this PR? It got moved to test ready, because there is another fix for the associated issue. So I disconnected it. Is this ready for review?

**Comment by kristinalim** [comment]
@mkllnk Yes, this can be tagged for Code Review. I wrote these tests to prove/understand the resulting permissions for variant overrides. Even after the bug fix of @luisramos0 in #2893, I think these tests would be handy to have in the code base.

I'll disconnect from the issue on Zenhub, update the PR title and description, and move to Code Review.

**Comment by luisramos0** [comment]
hey @kristinalim I fetched this branch, rebased and run the tests. it's all green.
I also had a look at the code, all good.
I edited the description to make it ready to go.
Looks good for review and merge to me.

**Comment by kristinalim** [comment]
Thanks for moving this forward @luisramos0! Somewhere along the way, I forgot this is already ready.

File: spec/models/spree/ability_spec.rb [hunk 1]
```diff
@@ -514,5 +514,89 @@ module Spree
...
+
+        describe "when the enterprise is not a distributor" do
+          let!(:authorized_enterprise) do
+            create(:enterprise, sells: "none").tap do |record|
+              create(:enterprise_relationship, parent: producer, child: record, permissions_list: [:create_variant_overrides])
+            end
+          end
+
+          it "should not have permission" do
+            should_not have_ability(manage_actions, for: variant_override)
```
Comment (sauloperez) [inline spec/models/spree/ability_spec.rb]: Could we use RSpec-3 syntax here? I know I sound boring and stubborn but sticking to this very old syntax makes it hard to stay-up-to-date. Let me know if you need help.

I'd like to automate this so I can avoid having to review syntax but I don't really know how to. Disabling `should` as RSpec's config allows is an all-or-nothing.

Comment (sauloperez) [reply spec/models/spree/ability_spec.rb]: ok, as you guys want. It's just that if we don't all get used to not writing RSpec2 syntax it'll make it harder.

**Comment by sigmundpetersen** [comment]
> What should we test?
>
> Build is green and all these new specs are green.

Merge?

**Comment by kristinalim** [comment]
@sigmundpetersen Sorry, I'll push a minor change based on the review of @sauloperez first! :slightly_smiling_face: Thank you so much!

File: spec/models/spree/ability_spec.rb (hunk 1 above)
Comment (kristinalim) [reply spec/models/spree/ability_spec.rb]: I rebased the branch and also changed to RSpec 2 syntax.

@sauloperez Sorry about that! This was an old PR, before I found out about `is_expected` from you. :slightly_smiling_face: Odd but I insisted on using `should` when dealing with `subject` until that time.

@luisramos0 I changed only the syntax in the tests I added, so no extra problems when merging.

**Comment by kristinalim** [comment]
This is good to go now.

File: spec/models/spree/ability_spec.rb (hunk 1 above)
Comment (luisramos0) [reply spec/models/spree/ability_spec.rb]: :+1: sorry, maybe I was too quick, my "no transpec until 2-0 merge" is not really valid for new tests...

**Pull request merged**